qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.upgrades;
+
+import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.xcontent.json.JsonXContent;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+
+public class FieldCapsIT extends AbstractRollingTestCase {
```

Comment:
could you add some javadocs to briefly explain the purpose of this test class? I would find it useful to explain what the expectations are and what we are testing.

Reply:
added in https://github.com/elastic/elasticsearch/pull/84455/commits/b87784582f014709fc92395f21d9649ea09dd7ee

Reply:
Sorry for being picky: could you expand a bit the javadocs you added with cases we are looking to cover and ashort explanation around setup choices you made? For instance, we have some common fields with same name but different type, some others with same name and same type, and some fields present only in one of the indices: what does that help us verify as part of this test? I was also wondering if it would help readability to split the test into multiple test methods.

Reply:
ok, I pushed https://github.com/elastic/elasticsearch/pull/84455/commits/c5fb7320eadb7bdaeb33fc21d25a7a9e178ae29a.

Reply:
thanks, looks great now!